scripts/todo_scanner.py

```
@@ -0,0 +1,460 @@
+            lines.append(f"\n{TEXT_INDENT}{file_report.path}")
+            for item in file_report.items:
+                lines.append(f"{FILE_INDENT}Line {item.line}: {item.content[:60]}")
+
+    return "\n".join(lines)
+
+
+def load_report(path: Path) -> dict[str, object]:
+    report_text = path.read_text(encoding=DEFAULT_ENCODING)
+    report_data = json.loads(report_text)
+    if not isinstance(report_data, dict):
+        raise ValueError(f"Report at {path} is not a JSON object")
```

Ruff (0.14.14)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)